src/k8s/pkg/k8sd/setup/kube_proxy.go

```diff
@@ -13,9 +13,14 @@ import (
 
 // KubeProxy configures kube-proxy on the local node.
 func KubeProxy(ctx context.Context, snap snap.Snap, hostname string, podCIDR string, extraArgs map[string]*string) error {
+	localhostAddress, err := utils.GetLocalhostAddress(podCIDR, "")
+	if err != nil {
+		return fmt.Errorf("failed to get locahost address: %w", err)
+	}
+
 	serviceArgs := map[string]string{
 		"--cluster-cidr":         podCIDR,
-		"--healthz-bind-address": "127.0.0.1",
+		"--healthz-bind-address": fmt.Sprintf("%s:10256", localhostAddress),
```

Comment:
nit: I'd prefer this to be determined once on the hook and passed as a parameter to the setup functions.

Reply:
fair point - updated